extensions/commands/art/cmd_promote_graph.py

```diff
@@ -0,0 +1,111 @@
+import json
+import os.path
+
+from conan.api.conan_api import ConanAPI
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command
+from conans.model.recipe_ref import RecipeReference
+from conans.model.package_ref import PkgReference
+from conan.api.model import MultiPackagesList
+from conan.errors import ConanException
+
+from utils import api_request, assert_server_or_url_user_password
+from cmd_server import get_url_user_password
+
+
+def _get_path_from_rrev(rrev):
+    recipe_ref = RecipeReference.loads(rrev)
+    rrev_path = f"/_/{recipe_ref.revision}" if recipe_ref.revision else ""
+    return f"_/{recipe_ref.name}/{recipe_ref.version}{rrev_path}"
+
+
+def _get_path_from_ref(ref):
+    try:
+        package_ref = PkgReference.loads(ref)
+        recipe_ref = package_ref.ref
+    except ConanException:
+        recipe_ref = RecipeReference.loads(ref)
+        package_ref = None
+
+    rrev_path = f"/_/{recipe_ref.revision}" if recipe_ref.revision else ""
+    pkgid_path = f"/package/{package_ref.package_id}" if package_ref and package_ref.package_id else ""
+    prev_path = f"/{package_ref.revision}" if package_ref and package_ref.revision else ""
+    return f"_/{recipe_ref.name}/{recipe_ref.version}{rrev_path}{pkgid_path}{prev_path}"
+
+
+def _request(url, user, password, request_type, request_url):
+    try:
+        return json.loads(api_request(request_type, f"{url}{request_url}", user, password))
+    except Exception as e:
+        raise ConanException(f"Error requesting {request_url}: {e}")
+
+
+def _promote_path(url, user, password, origin, destination, path):
+    ConanOutput().subtitle(f"Promoting {path}")
+    # The copy api creates a subfolder if the destination already exists, need to check beforehand to avoid this
+    try:
+        # This first request will raise a 404 if no file is found
+        _request(url, user, password, "get", f"api/storage/{destination}/{path}")
+        ConanOutput().warning("Destination already exists, skipping")
+    except ConanException:
+        _request(url, user, password, "post", f"api/copy/{origin}/{path}?to=/{destination}/{path}&suppressLayouts=0")
+        ConanOutput().success("Promoted file")
+
+
+@conan_command(group="Artifactory")
+def promote_graph(conan_api: ConanAPI, parser, *args):
+    """
+    Promote a pkglist file from an origin repository to a destination repository, without downloading the packages locally
+    """
+    parser.add_argument("origin", help="Artifactory origin repository.")
+    parser.add_argument("destination", help="Artifactory destination repository.")
```

Comment:
Maybe we want to process the destination from a Conan remote (from name, get the URL, and deduce the target from from it)?